File: plugin/src/main/java/dev/mmrl/Module.java
package dev.mmrl;

import android.webkit.JavascriptInterface;

import com.dergoogler.mmrl.platform.model.ModId;
import com.dergoogler.mmrl.webui.interfaces.WXInterface;
import com.dergoogler.mmrl.webui.interfaces.WXOptions;

public class Module extends WXInterface {
    public Module(WXOptions wxOptions) {
        super(wxOptions);
    }

    @Override
    public String getName() {
        return "module";
    }

    @JavascriptInterface
    public String getId() {
        return this.getModId().toString();
    }

    @JavascriptInterface
    public String getAdbDir() {
        ModId id = this.getModId();
        return ModId.Companion.getAdbDir(id).getPath();
    }

    @JavascriptInterface
    public String getConfigDir() {
        ModId id = this.getModId();
        return ModId.Companion.getConfigDir(id).getPath();
    }

    @JavascriptInterface
    public String getModuleConfigDir() {
        ModId id = this.getModId();
        return ModId.Companion.getModuleConfigDir(id).getPath();
    }

    @JavascriptInterface
    public String getModulesDir() {
        ModId id = this.getModId();
        return ModId.Companion.getModulesDir(id).getPath();
    }

    @JavascriptInterface
    public String getModuleDir() {
        ModId id = this.getModId();
        return ModId.Companion.getModuleDir(id).getPath();
    }

    @JavascriptInterface
    public String getWebRootDir() {
        ModId id = this.getModId();
        return ModId.Companion.getWebrootDir(id).getPath();
    }

    @JavascriptInterface
    public String getSystemDir() {
        ModId id = this.getModId();
        return ModId.Companion.getSystemDir(id).getPath();
    }

    @JavascriptInterface
    public String getPropFile() {
        ModId id = this.getModId();
        return ModId.Companion.getAdbDir(id).getPath();
    }
}

File: plugin/src/main/kotlin/dev/mmrl/Module.kt
package dev.mmrl

import android.webkit.JavascriptInterface
import com.dergoogler.mmrl.platform.model.ModId.Companion.adbDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.configDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.moduleConfigDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.moduleDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.modulesDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.propFile
import com.dergoogler.mmrl.platform.model.ModId.Companion.systemDir
import com.dergoogler.mmrl.platform.model.ModId.Companion.webrootDir
import com.dergoogler.mmrl.webui.interfaces.WXInterface
import com.dergoogler.mmrl.webui.interfaces.WXOptions

class Module(wxOptions: WXOptions) : WXInterface(wxOptions) {
    override var name: String = "module"

    @get:JavascriptInterface
    val id: String = modId.toString()

    @get:JavascriptInterface
    val adbDir: String = modId.adbDir.path

    @get:JavascriptInterface
    val configDir: String=modId.configDir.path

    @JavascriptInterface
    fun getModuleConfigDir(): String =modId.moduleConfigDir.path

//    @JavascriptInterface
//    fun getModuleConfigDir(vararg paths: String?): String {
//        val file = SuFile(moduleConfigDir, paths)
//        return file.getPath()
//    }

    @get:JavascriptInterface
    val modulesDir: String=modId.modulesDir.path

    @JavascriptInterface
    fun getModuleDir(): String = modId.moduleDir.path

//    @JavascriptInterface
//    fun getModuleDir(vararg paths: String?): String {
//        val file = SuFile(moduleDir, paths)
//        return file.getPath()
//    }

    @get:JavascriptInterface
    val webRootDir: String = modId.webrootDir.path

//    @JavascriptInterface
//    fun getWebRootDir(vararg paths: String?): String {
//        val file = SuFile(webrootDir, paths)
//        return file.getPath()
//    }

    @JavascriptInterface
    fun getSystemDir(): String = modId.systemDir.path

//    @JavascriptInterface
//    fun getSystemDir(vararg paths: String?): String {
//        val file = SuFile(systemDir, paths)
//        return file.getPath()
//    }

    @get:JavascriptInterface
    val propFile: String = modId.propFile.path
}
